src_c/line.c

```diff
@@ -490,6 +490,89 @@ pg_line_flip_ip(pgLineObject *self, PyObject *_null)
     Py_RETURN_NONE;
 }
 
+double 
+_lerp_helper(float start, float end, float amount) {
+    return (1 - amount) * start + amount * end;
+}
+
+static int
+_line_scale_helper(pgLineObject *line, double factor, double origin) {
```

Comment:
You have a lot of `line->line.` stuff in this function. You could simplify that by passing this function an actual `pgLineBase *`. That would make the function a bit easier to read (maybe a bit more efficient as well?).